packages/zoltan2/core/src/algorithms/color/Zoltan2_AlgHybridD1.hpp

```diff
@@ -0,0 +1,985 @@
+#define _ZOLTAN2_ALGHYBRIDGMB_HPP_
+
+#include <vector>
+#include <unordered_map>
+#include <iostream>
+#include <fstream>
+#include <queue>
+#include <sys/time.h>
+#include <algorithm>
+
+#include "Zoltan2_Algorithm.hpp"
+#include "Zoltan2_GraphModel.hpp"
+#include "Zoltan2_ColoringSolution.hpp"
+#include "Zoltan2_Util.hpp"
+#include "Zoltan2_TPLTraits.hpp"
+#include "Zoltan2_AlltoAll.hpp"
+
+#include "Tpetra_Core.hpp"
+#include "Teuchos_RCP.hpp"
+#include "Tpetra_Import.hpp"
+#include "Tpetra_FEMultiVector.hpp"
+
+#include "KokkosKernels_Handle.hpp"
+#include "KokkosKernels_IOUtils.hpp"
+#include "KokkosGraph_Distance1Color.hpp"
+#include "KokkosGraph_Distance1ColorHandle.hpp"
+#include <stdlib.h>
+
+//////////////////////////////////////////////
+//! \file Zoltan2_AlgHybridGMB.hpp
+//! \brief A hybrid version of the framework proposed by Gebremedhin and Manne
+
+namespace Zoltan2 {
+
+template <typename Adapter>
+class AlgDistance1 : public Algorithm<Adapter>
+{
+  public:
+  
+    using lno_t = typename Adapter::lno_t;
+    using gno_t = typename Adapter::gno_t;
+    using offset_t = typename Adapter::offset_t;
+    using scalar_t = typename Adapter::scalar_t;
+    using base_adapter_t = typename Adapter::base_adapter_t;
+    using map_t = Tpetra::Map<lno_t, gno_t>;
+    using femv_scalar_t = int;
+    using femv_t = Tpetra::FEMultiVector<femv_scalar_t, lno_t, gno_t>;
+    using device_type = Tpetra::Map<>::device_type;
+    using execution_space = Tpetra::Map<>::execution_space;
+    using memory_space = Tpetra::Map<>::memory_space;
+    double timer() {
+      struct timeval tp;
+      gettimeofday(&tp, NULL);
+      return ((double) (tp.tv_sec) + 1e-6 * tp.tv_usec);
+    }
+    
+  private:
+
+    void buildModel(modelFlag_t &flags); 
+
+    //function to invoke KokkosKernels distance-1 coloring    
+    template <class ExecutionSpace, typename TempMemorySpace, 
+              typename MemorySpace>
+    void colorInterior(const size_t nVtx, 
+                       Kokkos::View<lno_t*, Kokkos::Device<ExecutionSpace,MemorySpace> > adjs_view,
+                       Kokkos::View<offset_t*, Kokkos::Device<ExecutionSpace, MemorySpace> > offset_view, 
+                       Teuchos::RCP<femv_t> femv,
+		       Kokkos::View<lno_t*, Kokkos::Device<ExecutionSpace, MemorySpace> > vertex_list,
+		       size_t vertex_list_size = 0,
+                       bool recolor=false){
+      
+      //set up types to be used by KokkosKernels
+      using KernelHandle =  KokkosKernels::Experimental::KokkosKernelsHandle
+          <offset_t, lno_t, lno_t, ExecutionSpace, MemorySpace, 
+           MemorySpace>;
+      using lno_row_view_t = Kokkos::View<offset_t*, Kokkos::Device<ExecutionSpace, MemorySpace>>;
+      using lno_nnz_view_t = Kokkos::View<lno_t*, Kokkos::Device<ExecutionSpace, MemorySpace>>;
+
+      KernelHandle kh;
+      
+      //pick which KokkosKernels algorithm to use. Typically based on max graph degree.
```

Comment:
How is the KK algorithm chosen? I don't see any dependence on the max degree. This comment is confusing.

Reply:
The boolean is set based on max degree elsewhere, perhaps that elsewhere is where this comment belongs.

Reply:
I have revised this comment to explain more about why degree comes into play.